File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.py
def solution(targets):
    answer = 0
    # 정렬
    # 현재 끝점과 다음 시작점 비교, 이전이면 포함 & 다음 것도 비교, 아니면 미포함(answer+= 1) & 다음 행으로 넘어간다.
    targets.sort(key = lambda x:x[1])
    end = 0
    for s, e in targets:
        if s >= end:
            answer += 1
            end = e
    return answer

File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.js
function solution(targets) {
    let answer = 0;
    
    targets.sort((a, b) => a[1]-b[1])
    
    let end = 0
    for (let i = 0; i<targets.length; i++) {
        [s, e] = targets[i]
        
        if (s >= end) {
            end = e
            answer += 1   
        }
    }
    return answer;
}